activiti-core/activiti-process-validation/src/main/java/org/activiti/validation/validator/impl/FlowElementValidator.java

```diff
@@ -37,61 +39,76 @@
  */
 public class FlowElementValidator extends ProcessLevelValidator {
 
-	protected static final int ID_MAX_LENGTH = 255;
-
-	@Override
-	protected void executeValidation(BpmnModel bpmnModel, Process process, List<ValidationError> errors) {
-		for (FlowElement flowElement : process.getFlowElements()) {
-
-			if (flowElement instanceof Activity) {
-				Activity activity = (Activity) flowElement;
-				handleConstraints(process, activity, errors);
-				handleMultiInstanceLoopCharacteristics(process, activity, errors);
-				handleDataAssociations(process, activity, errors);
-			}
-
-		}
-
-	}
-
-	protected void handleConstraints(Process process, Activity activity, List<ValidationError> errors) {
-		if (activity.getId() != null && activity.getId().length() > ID_MAX_LENGTH) {
-			addError(errors, Problems.FLOW_ELEMENT_ID_TOO_LONG, process, activity,
-					"The id of a flow element must not contain more than " + ID_MAX_LENGTH + " characters");
-		}
-	}
-
-	protected void handleMultiInstanceLoopCharacteristics(Process process, Activity activity, List<ValidationError> errors) {
-		MultiInstanceLoopCharacteristics multiInstanceLoopCharacteristics = activity.getLoopCharacteristics();
-		if (multiInstanceLoopCharacteristics != null) {
-
-			if (StringUtils.isEmpty(multiInstanceLoopCharacteristics.getLoopCardinality())
-	    		&& StringUtils.isEmpty(multiInstanceLoopCharacteristics.getInputDataItem())) {
-
-			  addError(errors, Problems.MULTI_INSTANCE_MISSING_COLLECTION, process, activity,
-	    			"Either loopCardinality or loopDataInputRef/activiti:collection must been set");
-	    }
-
-		}
-	}
-
-	protected void handleDataAssociations(Process process, Activity activity, List<ValidationError> errors) {
-	  if (activity.getDataInputAssociations() != null) {
-	  	for (DataAssociation dataAssociation : activity.getDataInputAssociations()) {
-	  		if (StringUtils.isEmpty(dataAssociation.getTargetRef())) {
-	  			 addError(errors, Problems.DATA_ASSOCIATION_MISSING_TARGETREF, process, activity,
-	  					 "Targetref is required on a data association");
-	  	    }
-	  	}
-	  }
-	  if (activity.getDataOutputAssociations() != null) {
-	  	for (DataAssociation dataAssociation : activity.getDataOutputAssociations()) {
-	  		if (StringUtils.isEmpty(dataAssociation.getTargetRef())) {
-	  			 addError(errors, Problems.DATA_ASSOCIATION_MISSING_TARGETREF, process, activity,
-	  					 "Targetref is required on a data association");
-	  	    }
-	  	}
-	  }
+    protected static final int ID_MAX_LENGTH = 255;
+    @Value("${loop.cardinality.threshold}")
```

Comment:
add a blank new line above, prefix it with acitivi and specify the default value here.
```suggestion

    @Value("${activiti.loop.cardinality.threshold:200}")
```